### DISCUSSION TITLE
Add support for Google Error Handling model

### DISCUSSION THREAD

**Discussion by childsev** [timostamm/protobuf-ts#516] [Ideas]
Google has a rich error handling model which supports embedding custom error details in RpcError metadata: https://grpc.io/docs/guides/error/#richer-error-model. 

Any there currently any plans to support this as part of Protobuf-ts? 

I've been able to verify that the error details are available by base64 decoding the included metadata field for "grpc-status-details-bin" using a custom RPC interceptor, so it appears to be possible to unpack err details on the client side

```
const META_GRPC_STATUS_FIELD = "grpc-status-details-bin";

const onrejected: (reason: RpcError) => (PromiseLike<RpcError> | RpcError) = (reason) => {
    console.log(`ErrorInterceptor interceptor unary call error: ${method.service.typeName} / ${method.name}`);

    const meta = reason.meta;
    if (meta?.[META_GRPC_STATUS_FIELD]) {
        const base64EncodedStatus = meta[META_GRPC_STATUS_FIELD];
        if (Array.isArray(base64EncodedStatus)) {
            console.log("Is array")
        } else {
            const decoded = Buffer.from(base64EncodedStatus, 'base64');

            meta[META_GRPC_STATUS_FIELD] = decoded.toString();

            // TODO: Unmarshal to the protobuf type
            // Need to include google.protobuf.Any proto
            // and the corresponding error details
            // https://github.com/timostamm/protobuf-ts/blob/3a5b9f982d30d3422d856176d822bed027e981e4/packages/test-fixtures/google/protobuf/any.proto#L125
        }
    }
    return Promise.reject(reason);
}
```

**Comment by timostamm** [comment]
Hey Evan, once you have the base64-decoded data from the `grpc-status-details-bin` field, you can parse it as a [google.rpc.Status](https://github.com/grpc/grpc/blob/master/src/proto/grpc/status/status.proto) message with `Status.fromBinary()`. Then you unpack the error details from the repeated google.protobuf.Any. Also see https://github.com/timostamm/protobuf-ts/issues/370.

protobuf-ts uses @grpc/grpc-js under the hood for gRPC, and it doesn't offer any support for error details, unfortunately.
